src/Avalonia.Base/Input/GestureRecognizers/PinchGestureRecognizer.cs

```diff
@@ -2,7 +2,7 @@
 
 namespace Avalonia.Input
 {
-    public class PinchGestureRecognizer : StyledElement, IGestureRecognizer
+    public class PinchGestureRecognizer : AvaloniaObject, IGestureRecognizer
```

Comment:
Recognizer being a StyledElement is required for TemplateBinding to work

https://github.com/AvaloniaUI/Avalonia/blob/23d96c73667091bad8c28e19c79bab82bbbfb211/src/Avalonia.Base/Input/GestureRecognizers/GestureRecognizerCollection.cs#L33-L41

Reply:
`ScrollGestureRecognizer` was converted to not be a `StyledElement` in https://github.com/AvaloniaUI/Avalonia/commit/e3c6ee0da3ee703e32c956d1521a37a7e0c86165 and I think the approach there is less hacky.

Reply:
Though actually, I did have to change it here because it wasn't working: https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/ScrollViewer.xaml#L42